db/system_distributed_keyspace.cc

```diff
@@ -335,17 +337,33 @@ static set_type_impl::native_type prepare_cdc_streams(const std::vector<cdc::str
     return ret;
 }
 
+// cdc_streams_descriptions mutation
+static future<mutation> get_mutation(
+        const database& db,
+        db_clock::time_point time,
+        const std::vector<cdc::stream_id>& streams) {
+    auto s = db.find_schema(system_distributed_keyspace::NAME, system_distributed_keyspace::CDC_DESC);
+    mutation m(s, partition_key::from_singular(*s, time));
+    m.set_cell(clustering_key::make_empty(), to_bytes("streams"),
+            make_set_value(cdc_streams_set_type, prepare_cdc_streams(streams)), api::new_timestamp());
+    co_return m;
+}
+
 future<>
 system_distributed_keyspace::create_cdc_desc(
         db_clock::time_point time,
         const std::vector<cdc::stream_id>& streams,
         context ctx) {
-    return _qp.execute_internal(
-            format("INSERT INTO {}.{} (time, streams) VALUES (?,?)", NAME, CDC_DESC),
+    using namespace std::chrono_literals;
+    auto m = co_await get_mutation(_qp.db(), time, streams);
```

Comment:
I have to admit that my coroutine knowledge is quite poor but why do we need co_await for a call that never blocks?

Reply:
`get_mutation` returns a `future`, we want to wait on this future. We don't want to discard it.
Perhaps I don't fully understand the question

Reply:
but get_mutation never blocks so there's no need to wait. The correct answer is that in the following commits you make get_mutation do some calls that could block.

Reply:
So the correct question is "why `get_mutation` returns a future"
which you answered correctly indeed.